src/runtime_src/core/tools/xbmgmt2/flash/firmware_image.cpp

```diff
@@ -753,7 +753,7 @@ firmwareImage::firmwareImage(const std::string& file, imageType type) :
     }
     else
     {
-        if (type != MCS_FIRMWARE_PRIMARY)
+        if (type != BMC_FIRMWARE && type != MCS_FIRMWARE_PRIMARY)
```

Comment:
```suggestion
        if ((type != BMC_FIRMWARE) && (type != MCS_FIRMWARE_PRIMARY))
```
Just makes your intention that more obvious.